apps/src/applab/commands.js

```diff
@@ -866,6 +874,13 @@ applabCommands.setImageURL = function (opts) {
   var element = document.getElementById(opts.elementId);
   if (divApplab.contains(element) && element.tagName === 'IMG') {
     element.src = Applab.maybeAddAssetPathPrefix(opts.src);
+
+    if (!ensureCached[element.src]) {
+      ensureCached[element.src] = true;
+      var img = new Image();
+      img.src = element.src;
```

Comment:
I might move line 879 to be after 881, just to make what's happening a little more obvious (we're loading, then marking as cached).

Also, could we run into any problems where we claim we're cached, but haven't actually loaded the image yet (i.e. after  setting img.src/ensureCached, but before img.onload is called, or if img.onload fails).